src/admin_api.ts

```diff
@@ -0,0 +1,374 @@
+      const basePath = 'admin/users';
+      const queryString = buildQueryPath({
+        filter: conditions,
+        metadata,
+        page: params,
+      });
+  
+      return await this.http.adminRequest(`${basePath}${queryString}`, 'GET');
+    }
+  
+    /**
+     * Register a new xPub
+     *
+     * @param {string} rawXPub - Raw xPub key to register
+     * @param {Metadata} metadata - Metadata for the xPub
+     * @returns {Promise<XPub>} Registered xPub information
+     */
+    async createXPub(rawXPub: string, metadata: Metadata): Promise<XPub> {
+      return await this.http.adminRequest('admin/users', 'POST', {
+        key: rawXPub,
+        metadata,
+      });
+    }
+  
+    /**
+     * Get a paymail by address
+     *
+     * @param {string} id - Paymail ID or address
+     * @returns {Promise<PaymailAddress>} Paymail information
+     */
+    async paymail(id: string): Promise<PaymailAddress> {
+      return await this.http.adminRequest(`admin/paymails/${id}`, 'GET');
+    }
+  
+    /**
+     * Get a list of all paymails
+     *
+     * @param {AdminPaymailFilter} conditions - Filter conditions for paymails
+     * @param {Metadata} metadata - Metadata filter
+     * @param {QueryPageParams} params - Pagination parameters
+     * @returns {Promise<PageModel<PaymailAddress>>} List of paymail addresses
+     */
+    async paymails(
+      conditions: AdminPaymailFilter,
+      metadata: Metadata,
+      params: QueryPageParams,
+    ): Promise<PageModel<PaymailAddress>> {
+      const basePath = 'admin/paymails';
+      const queryString = buildQueryPath({
+        metadata,
+        page: params,
+        filter: conditions,
+      });
+  
+      return await this.http.adminRequest(`${basePath}${queryString}`, 'GET');
+    }
+  
+    /**
+     * Create a new paymail
+     *
+     * @param {string} rawXPub - Raw xpub to register the paymail to
+     * @param {string} address - Paymail address (e.g., alias@domain.com)
+     * @param {string} publicName - Public name for the paymail
+     * @param {string} avatar - Avatar URL
+     * @param {Metadata} metadata - Additional metadata
+     * @returns {Promise<PaymailAddress>} Created paymail address
+     */
+    async createPaymail(
+      rawXPub: string,
+      address: string,
+      publicName: string,
+      avatar: string,
+      metadata: Metadata,
+    ): Promise<PaymailAddress> {
+      return await this.http.adminRequest('admin/paymails', 'POST', {
+        metadata,
+        key: rawXPub,
+        address,
+        publicName,
+        avatar,
+      });
+    }
+  
+    /**
+     * Delete a paymail
+     */
+    async deletePaymail(id: string, address: string): Promise<void> {
+      await this.http.adminRequest(`admin/paymails/${id}`, 'DELETE', { address });
+    }
+  
+    /**
+     * Get webhook subscriptions
+     *
+     * @returns {Promise<Webhook[]>} List of webhook subscriptions
+     */
+    async getWebhooks(): Promise<Webhook[]> {
```

Comment:
to meet the convention in method names here, it shouldn't have `get` prefix in name
 ```suggestion
    async webhooks(): Promise<Webhook[]> {
```

Reply:
No problem, will update. Thanks